src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -152,27 +159,178 @@ public void AddPackageReference(string projectPath, LibraryDependency libraryDep
+
+            // Save the updated props file.
+            directoryBuildPropsRootElement.Save();
+        }
+
+        /// <summary>
+        /// Get the Directory build props root element for projects onboarded to CPM.
+        /// </summary>
+        /// <param name="project">Project that needs to be modified.</param>
+        /// <returns>The directory build props root element.</returns>
+        private ProjectRootElement GetDirectoryBuildPropsRootElement(Project project)
+        {
+            // Get the Directory.Packages.props path.
+            string directoryPackagesPropsPath = project.GetPropertyValue(DirectoryPackagesPropsPathPropertyName);
+            ProjectRootElement directoryBuildPropsRootElement = project.Imports.FirstOrDefault(i => i.ImportedProject.FullPath.Equals(directoryPackagesPropsPath)).ImportedProject;
+            return directoryBuildPropsRootElement;
+        }
+
+        /// <summary>
+        /// Add package name and version into the props file.
+        /// </summary>
+        /// <param name="project">Project that is being modified.</param>
+        /// <param name="itemGroup">Item group that needs to be modified in the props file.</param>
+        /// <param name="libraryDependency">Package Dependency of the package to be added.</param>
+        private void AddPackageVersionIntoPropsItemGroup(Project project, ProjectItemGroupElement itemGroup,
+            LibraryDependency libraryDependency)
+        {
+            // Add both package reference information and version metadata using the PACKAGE_VERSION_TYPE_TAG.
+            var item = itemGroup.AddItem(PACKAGE_VERSION_TYPE_TAG, libraryDependency.Name);
+            var packageVersion = AddVersionMetadata(libraryDependency, item);
+
+            Logger.LogInformation(string.Format(CultureInfo.CurrentCulture,
+            Strings.Info_AddPkgAdded,
+            libraryDependency.Name,
+            packageVersion,
+            itemGroup.ContainingProject.FullPath
+            ));
+
+            AddExtraMetadataToProjectItemElement(libraryDependency, item);
+        }
+
+        /// <summary>
+        /// Add package name and version into item group.
+        /// </summary>
+        /// <param name="itemGroup">Item group to add to.</param>
+        /// <param name="libraryDependency">Package Dependency of the package to be added.</param>
+        private void AddPackageReferenceIntoItemGroup(ProjectItemGroupElement itemGroup,
+            LibraryDependency libraryDependency)
+        {
+            // Add both package reference information and version metadata using the PACKAGE_REFERENCE_TYPE_TAG.
+            var item = itemGroup.AddItem(PACKAGE_REFERENCE_TYPE_TAG, libraryDependency.Name);
+            var packageVersion = AddVersionMetadata(libraryDependency, item);
+
+            Logger.LogInformation(string.Format(CultureInfo.CurrentCulture, Strings.Info_AddPkgAdded, libraryDependency.Name, packageVersion, itemGroup.ContainingProject.FullPath));
```

Comment:
My current implementation does do that. This actually has not changed from the original implementation since this function is only called in non-CPM cases when the package reference must be updated. After the item is added to the Item Group and the version metadata is added in the .proj file, the information is logged. Does this align with what you were looking for?

Reply:
To make it more clear, I logged information after all metadata is added.